roles/edpm_reboot/tasks/needsreboot.yaml

```diff
@@ -0,0 +1,80 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Make sure yum-utils is installed
+  become: true
+  ansible.builtin.dnf:
+    name: yum-utils
+
+- name: Check if reboot is required with needs-restarting
+  ansible.builtin.command: needs-restarting -r
+  register: needs_restarting_output
+  changed_when: false
+  failed_when: needs_restarting_output.rc >= 2
+
+- name: Print return information from needs-restarting
+  ansible.builtin.debug:
+    var: needs_restarting_output.stdout
+
+- name: Create reboot file for needs-restarting
```

Comment:
I find this sequence somewhat confusing. It's a block of 3 tasks to produce a file which could have been done in a single shell task at line 22 above. At scale, this extends the run-time significantly. Are we approaching how we do things in a way that scales well?